File: src/main/java/me/xii69/justonemore/JustOneMore.java
package me.xii69.justonemore;

import com.velocitypowered.api.event.Subscribe;
import com.velocitypowered.api.event.proxy.ProxyPingEvent;
import com.velocitypowered.api.plugin.Plugin;
import com.velocitypowered.api.proxy.server.ServerPing;

@Plugin(
        id = "justonemore",
        name = "JustOneMore",
        version = "1.1.0",
        description = "JustOneMore plugin for Velocity",
        authors = {"xii69"}
)

public class JustOneMore {

    @Subscribe
    public void onProxyPing(ProxyPingEvent event) {
        final ServerPing serverPing = event.getPing();
        if (serverPing.getPlayers().isPresent()) {
            final ServerPing.Builder builder = serverPing.asBuilder();
            final int onlinePlayers = builder.getOnlinePlayers();
            event.setPing(builder.maximumPlayers(onlinePlayers + 1).build());
        }
    }
}

File: src/main/java/me/xii69/justonemore/velocity/JustOneMore.kt
package me.xii69.justonemore.velocity

import com.google.inject.Inject
import com.velocitypowered.api.event.PostOrder
import com.velocitypowered.api.event.Subscribe
import com.velocitypowered.api.event.proxy.ProxyInitializeEvent
import com.velocitypowered.api.event.proxy.ProxyPingEvent
import me.xii69.justonemore.shared.Metrics.Factory

class JustOneMore @Inject constructor(
    private val metricsFactory: Factory
) {
    @Subscribe
    fun onProxyInitialization(event: ProxyInitializeEvent) = metricsFactory.make(this, 22308)

    @Subscribe(order = PostOrder.LAST)
    fun onProxyPing(event: ProxyPingEvent) {
        if (event.ping.players.isPresent) event.ping = event.ping.asBuilder().maximumPlayers(event.ping.asBuilder().onlinePlayers + 1).build()
    }
}
